common/src/node.rs

```diff
@@ -125,24 +154,30 @@ impl ChunkLayout {
     }
 
     /// Converts a single coordinate from dual coordinates in the Klein-Beltrami model to an integer coordinate
-    /// suitable for voxel lookup. Margins are included. Returns `None` if the coordinate is outside the chunk.
+    /// suitable for voxel lookup. Returns `None` if the coordinate is outside the chunk.
     #[inline]
-    pub fn dual_to_voxel(&self, dual_coord: f32) -> Option<usize> {
+    pub fn dual_to_voxel(&self, dual_coord: f32) -> Option<u8> {
         let floor_grid_coord = (dual_coord * self.dual_to_grid_factor).floor();
 
         if !(floor_grid_coord >= 0.0 && floor_grid_coord < self.dimension as f32) {
             None
         } else {
-            Some(floor_grid_coord as usize + 1)
+            Some(floor_grid_coord as u8)
         }
     }
 
     /// Converts a single coordinate from grid coordinates to dual coordiantes in the Klein-Beltrami model. This
     /// can be used to find the positions of voxel gridlines.
     #[inline]
-    pub fn grid_to_dual(&self, grid_coord: usize) -> f32 {
+    pub fn grid_to_dual(&self, grid_coord: u8) -> f32 {
         grid_coord as f32 / self.dual_to_grid_factor
     }
+
+    /// Takes in a single grid coordinate and returns a range containing all voxel coordinates surrounding it.
+    #[inline]
+    pub fn neighboring_voxels(&self, grid_coord: u8) -> impl Iterator<Item = u8> {
+        grid_coord.saturating_sub(1)..(grid_coord + 1).min(self.dimension())
```

Comment:
It can't hurt. There might be other issues (besides memory/performance) if the `grid_coord` could be 255, but we can at least be robust about it here.